api/plugin_helpers.go

```diff
@@ -120,7 +124,7 @@ func VaultPluginTLSProvider(apiTLSConfig *TLSConfig) func() (*tls.Config, error)
 // VaultPluginTLSProviderContext is run inside a plugin and retrieves the response
 // wrapped TLS certificate from vault. It returns a configured TLS Config.
 func VaultPluginTLSProviderContext(ctx context.Context, apiTLSConfig *TLSConfig) func() (*tls.Config, error) {
-	if os.Getenv(PluginMetadataModeEnv) == "true" {
+	if os.Getenv(PluginAutoMTLSEnv) == "true" || os.Getenv(PluginMetadataModeEnv) == "true" {
```

Comment:
Yes, go-plugin [defines AutoMTLS](https://github.com/hashicorp/go-plugin/blob/a02e5c7a911db823ae167d00bc1024f99d340258/client.go#L201-L203) as mutually exclusive to setting a TLSConfig/TLSProvider. If the TLSProvider is set, go-plugin will [not configure AutoMTLS on the plugin server](https://github.com/hashicorp/go-plugin/blob/a02e5c7a911db823ae167d00bc1024f99d340258/server.go#L285-L298). So here we return so that AutoMTLS is configured for both the go-plugin client and server. The `PluginAutoMTLSEnv` will only be set for V5 plugins.